src/plugins/event-plugins/DomEventPlugin.ts

```diff
@@ -129,13 +129,30 @@ export class DomEventPlugin implements Plugin {
             };
             if (cssLocator !== undefined) {
                 eventData.cssLocator = cssLocator;
+                const elementRumId = this.getElementRumId(event, cssLocator);
+                if (elementRumId) {
+                    eventData.elementRumId = elementRumId;
```

Comment:
issue: We'll need to add `elementRumId` (or simply `rumId`) to the `dom-event.json` schema, or it won't be ingested.

Reply:
Right now we have data from two different sources going into the `elementId` field. I wonder if we should disambiguate these by (1) making `elementId` optional and only add it if an  was found, and (2) adding an `element` field for the type of element.

The event would look like:
```
{
  event: 'click',
  element: 'SPAN',
  cssLocator: 'button',
  rumId: 'button-12345'
}
```

Reply:
This is a good suggestion. I agree to both.

Reply:
> issue: We'll need to add `elementRumId` (or simply `rumId`) to the `dom-event.json` schema, or it won't be ingested.

It is added to that file. And I will rename it as rumId, it looks simpler and better.